pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/DataDictionaryExporter.java

```diff
@@ -0,0 +1,293 @@
+package org.broadinstitute.dsm.model.elastic.export.tabular;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.util.CellRangeAddress;
+import org.apache.poi.xssf.streaming.SXSSFCell;
+import org.apache.poi.xssf.streaming.SXSSFRow;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import spark.Response;
+
+/** Writes a data dictionary file based on the given configs.
+ * columns include variable name, type, description, and options
+ */
+public class DataDictionaryExporter extends ExcelParticipantExporter {
+    private static final int ROW_ACCESS_WINDOW_SIZE = 200;
+    private int currentRowNum = -1;
+    private static final String SHEET_NAME = "Data dictionary";
+    private static final String FILE_NAME = "DataDictionary.xlsx";
+
+    private final CellStyle wrapStyle;
+    private final CellStyle boldStyle;
+    private final CellStyle boldUnderlineStyle;
+    private static final Map<String, String> QUESTION_TYPE_TO_DATA_TYPE = Map.of(
+            "DATE", "datetime",
+            "DATE_SHORT", "date",
+            "AGREEMENT", "boolean",
+            "BOOLEAN", "boolean",
+            "NUMERIC", "number",
+            "CHECKBOX", "boolean",
+            "NUMBER", "number"
+    );
+
+
+    public String getExportFilename() {
+        return FILE_NAME;
+    }
+
+    /**
+     * initializes the dictionary and internal spreadsheet
+     */
+    public DataDictionaryExporter(List<ModuleExportConfig> configs) {
+        super(configs, null, TabularParticipantExporter.XLSX_FORMAT);
+        wrapStyle = workbook.createCellStyle();
+        wrapStyle.setWrapText(true);
+        boldStyle = workbook.createCellStyle();
+        Font boldFont = workbook.createFont();
+        boldFont.setBold(true);
+        boldStyle.setFont(boldFont);
+
+        boldUnderlineStyle = workbook.createCellStyle();
+        Font boldUnderlineFont = workbook.createFont();
+        boldUnderlineFont.setBold(true);
+        boldUnderlineFont.setUnderline(Font.U_SINGLE);
+        boldUnderlineStyle.setFont(boldUnderlineFont);
+    }
+
+    /** writes the dictionary */
+    public void export(OutputStream os) throws IOException {
+        sheet.setColumnWidth(0, 40 * 256);
+        sheet.setColumnWidth(1, 10 * 256);
+        sheet.setColumnWidth(2, 12 * 256);
+        sheet.setColumnWidth(3, 60 * 256);
+        sheet.setColumnWidth(4, 40 * 256);
```

Comment:
is it fixed size columns number? if so I would extract 0, 1, 2, 3, 4, 5 so called magic numbers as constants with meaningful name like 0 -> variable name, 1 - data type, etc